addon/components/polaris-text-field/resizer.js

```diff
@@ -0,0 +1,128 @@
+import Component from '@ember/component';
+import { htmlSafe } from '@ember/string';
+import { computed } from '@ember/object';
+import { next } from '@ember/runloop';
+import ContextBoundTasksMixin from 'ember-lifeline/mixins/run';
+import ContextBoundEventListenersMixin from 'ember-lifeline/mixins/dom';
+import layout from '../../templates/components/polaris-text-field/resizer';
+
+const REPLACE_REGEX = /[\n&<>]/g;
+
+const ENTITIES_TO_REPLACE = {
+  '&': '&amp;',
+  '<': '&lt;',
+  '>': '&gt;',
+  '\n': '<br>',
+};
+
+function replaceEntity(entity) {
+  return ENTITIES_TO_REPLACE[entity] || entity;
+}
+
+export default Component.extend(ContextBoundTasksMixin, ContextBoundEventListenersMixin, {
+  classNames: ['Polaris-TextField__Resizer'],
+  attributeBindings: ['ariaHidden:aria-hidden'],
+
+  ariaHidden: 'true',
+
+  layout,
+
+  /**
+   * The value of the textarea
+   *
+   * @property contents
+   * @public
+   * @type {String}
+   * @default null
+   */
+  contents: null,
+
+  /**
+   * The height (in px) of the textarea
+   *
+   * @property currentHeight
+   * @public
+   * @type {Number}
+   * @default null
+   */
+  currentHeight: null,
+
+  /**
+   * The multiline value of the textarea if
+   * a numeric value was passed-in to the polaris-text-field
+   *
+   * @property minimumLines
+   * @public
+   * @type {Number}
+   * @default null
+   */
+  minimumLines: null,
+
+  /**
+   * Callback when the height of the resize container changes
+   *
+   * @property onHeightChange
+   * @public
+   * @type {Function}
+   * @default noop
+   */
+  onHeightChange(/* height */) {},
+
+  finalContents: computed('contents', function() {
+    let contents = this.get('contents');
+
+    contents = contents
+      ? `${ contents.replace(REPLACE_REGEX, replaceEntity) }<br>`
+      : '<br>';
+
+    return htmlSafe(contents);
+  }).readOnly(),
+
+  contentsForMinimumLines: computed('minimumLines', function() {
+    let minimumLines = this.get('minimumLines');
+    let content = '';
+
+    for (let line = 0; line < minimumLines; line++) {
+      content = `${ content }<br>`;
+    }
+
+    return htmlSafe(content);
+  }).readOnly(),
+
+  handleHeightCheck() {
+    let [ contentNode, minimumLinesNode ] = this.element.querySelectorAll('.Polaris-TextField__DummyInput');
+
+    if (!contentNode || !minimumLinesNode) {
+      return;
+    }
+
+    let contentHeight = contentNode.offsetHeight;
+    let minimumHeight = minimumLinesNode.offsetHeight;
+    let newHeight = Math.max(contentHeight, minimumHeight);
+    let { currentHeight, onHeightChange } = this.getProperties('currentHeight', 'onHeightChange');
+
+    if (newHeight !== currentHeight) {
+      next(this, () => {
```

Comment:
I'd imagine the flow is something like this:

- `resizer` calls `onHeightChange`
- parent `text-field` component handles `onHeightChange`, which updates the `height` property
- parent `text-field` component passes updated `height` property into `resizer` component
- `resizer` component's `didUpdateAttrs` hook ends up invoking `onHeightChange` again because `currentHeight` hasn't updated yet, taking us around the loop a second time...

So the way to handle this would be to stop `onHeightChange` running more than once (especially in a single run loop). For that reason, `scheduleOnce` is probably more appropriate than `next` I'd say...